cmake/modules/BuildDPDK.cmake

```diff
@@ -181,7 +181,8 @@ function(do_export_dpdk dpdk_dir)
   set_target_properties(dpdk::dpdk PROPERTIES
     INTERFACE_INCLUDE_DIRECTORIES ${DPDK_INCLUDE_DIR}
     INTERFACE_LINK_LIBRARIES
-    "-Wl,--whole-archive $<JOIN:${DPDK_ARCHIVES}, > -Wl,--no-whole-archive ${dpdk_numa} -Wl,-lpthread,-ldl")
+    "-Wl,--whole-archive $<JOIN:${DPDK_ARCHIVES}, > -Wl,--no-whole-archive ${dpdk_numa} -Wl,-lpthread,-ldl,-lmlx5")
+  target_link_libraries(dpdk::dpdk INTERFACE RDMA::RDMAcm IBVerbs::verbs)
```

Comment:
instead of creating a new PR, would be better to force-push to the previous branch from which you created the PR that i commented on, so that we can preserve the discussion history

Reply:
OK, I got it. I will pay attention to it next time.

Since I have deleted the previous branch, I can no longer push on the previous branch. 
Can we start the discussion based on this branch?

Reply:
Sure. But the same issues remain. 

- How to reproduce the problem you are addressing?
- Why is the RDMA linkage added to DPDK?

Reply:
1. Add rdma compilation options in the BuildSPDK.cmake file, eg:
foreach(c lvol env_dpdk sock rdma nvmf bdev nvme conf thread trace
 ...
CONFIGURE_COMMAND ./configure
--with-dpdk=${DPDK_DIR}
--with-rdma=mlx5_dv
--without-isal

Then use ninja to compile ceph, and a link error will be reported.
2、Since DPDK is needed to compile spdk,  add RDMA linkage to DPDK is work.

Reply:
i don't think this is an ideal workflow. if one is expected to edit `BuildSPDK.cmake` manually, he/she should be able to edit `BuildDPDK.cmake` as well.

Reply:
But if don't edit BuildSPDK.cmake manually, how should ceph let spdk enable rdma?

Reply:
i think there are couple approaches

1. enable ceph to build with external SPDK
2. add a cmake option to build spdk with rdma
3. fix spdk's buiding system so that it pulls in the used libraries. i suggested this in the pull request that you closed.

Reply:
OK, I will try your suggestion.

Reply:
hi, @tchaikov I resubmitted the patch according to your suggestion. Can you review it when you are available?

Reply:
> i think there are couple approaches
> 
> 1. enable ceph to build with external SPDK
> 2. add a cmake option to build spdk with rdma
> 3. fix spdk's buiding system so that it pulls in the used libraries. i suggested this in the pull request that you closed.

Hi @tchaikov can you review this PR please